### PR TITLE
[Emacs Lisp] Emacs Lisp concepts

### PR DESCRIPTION
**Pull request by porkostomus**
- added a working Concepts list and mermaid chart
- added a `_sidebar.md` and created the relevant docs
- added myself as a maintainer
- updated track progress on the main `README.md`
- added `config.json`

### CODE REVIEW

File: languages/emacs-lisp/reference/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- Special forms
+- Atoms
+- Quote
+- S-expressions
+- [Evaluation][evaluation]
+- [Functions][functions]
+- [Variables][variables]
+- [Macros][macros]
+- Side-effects
+- Arguments
```
Comment (maurelio1234) [inline languages/emacs-lisp/reference/README.md]: what do you mean here precisely? the arguments we pass to a function? it's weird that we do not have them on the toplevel list of concepts

File: languages/emacs-lisp/reference/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+- Special forms
+- Atoms
+- Quote
+- S-expressions
+- [Evaluation][evaluation]
+- [Functions][functions]
+- [Variables][variables]
+- [Macros][macros]
+- Side-effects
```
Comment (maurelio1234) [inline languages/emacs-lisp/reference/README.md]: what do you mean here?

File: languages/emacs-lisp/reference/progression.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Emacs Lisp Concept Exercise Progression
+
+This is a working document to keep track of ideas and thoughts on how the progression through the Concept Exercises on the Emacs Lisp track could work.
+
+```mermaid
```
Comment (maurelio1234) [inline languages/emacs-lisp/reference/progression.md]: is it normal that this is not rendered?

File: languages/emacs-lisp/reference/README.md (hunk 2 above)
Comment (porkostomus) [reply languages/emacs-lisp/reference/README.md]: About side effects? The example given in the intro doc is the `set` function.

File: languages/emacs-lisp/reference/README.md (hunk 1 above)
Comment (porkostomus) [reply languages/emacs-lisp/reference/README.md]: Yeah that should be part of functions which I believe is part of lists. I'll fix the order. Arguments, meaning every item that is passed to the function yes.

File: languages/emacs-lisp/reference/README.md (hunk 2 above)
Comment (maurelio1234) [reply languages/emacs-lisp/reference/README.md]: It looks close to the concept of `variable` itself.

```This is the concept of assigning a name to a piece of data.``` (from `reference/concepts`)

File: languages/emacs-lisp/reference/progression.md (hunk 3 above)
Comment (porkostomus) [reply languages/emacs-lisp/reference/progression.md:5]: It's rendered on the website, which can be [run locally](https://github.com/exercism/v3/blob/master/docs/maintainers/website.md).

**Pull request merged**